lib/SILGen/SILGenPoly.cpp

```diff
@@ -3746,8 +3761,8 @@ SILGenModule::getOrCreateAutoDiffAssociatedFunctionThunk(
 
   SmallVector<SILValue, 8> directResults;
   extractAllElements(apply, thunkSGF.B, directResults);
-  auto linearMap = directResults.back();
-  auto linearMapFnType = linearMap->getType().castTo<SILFunctionType>();
+  auto linearMap = ManagedValue::forBorrowedObjectRValue(directResults.back());
+  auto linearMapFnType = linearMap.getType().castTo<SILFunctionType>();
   auto targetLinearMapFnType = thunk->mapTypeIntoContext(
       origAssocFnType->getResults().back().getSILStorageType())
           .castTo<SILFunctionType>();
```

Comment:
I just noticed that you are doing manual memory management below in SILGen. We should use SILGen ManagedValue operations instead. Please file an issue to track this future work.

Reply:
I have a WIP patch after this one that enables ownership for SILGen linear map thunks.